src/connection/host_lookup.c

```diff
@@ -0,0 +1,435 @@
+  fp = fopen (hosts_conf_dir, "r");
+  if (fp == NULL)
+    {
+      return -1;
+      //error msg-> temp
+    }
+
+  while (fgets (file_line, HOSTNAME_BUF_SIZE, fp) != NULL)
+    {
+//verifying duplicated hostname will be implemented soon
+      if (file_line[0] == '#')
+	continue;
+      if (file_line[0] == '\n')
+	continue;
+
+      token = strtok_r (file_line, delim, &save_ptr_strtok);
+      storage_flag = IPADDR_STORE;
+
+      char map_ipaddr[HOSTNAME_BUF_SIZE];
+      char map_hostname[HOSTNAME_BUF_SIZE];
+
+      do
+	{
+	  if (*token == '\0' || *token == '#')
+	    {
+	      break;
+	    }
+	  if (!storage_flag)
+	    {
+	      strcpy (hostent_List[host_count].ipaddr, token);
+	      strcpy (map_ipaddr, token);
+
+	      storage_flag = HOSTNAME_STORE;
+	    }
+	  else
+	    {
+	      strcpy (hostent_List[host_count].hostname, token);
+	      strcpy (map_hostname, token);
+	    }
+	}
+      while (token = strtok_r (NULL, delim, &save_ptr_strtok));
+//naming rule check, detailed rule, error set
+//if hostent_List[host_count].hostname is empty, set error
+      if (hostent_List[host_count].hostname != NULL)
+	{
+	  if ((user_host_Map.find (map_hostname) == user_host_Map.end ()))
+	    {
+	      user_host_Map[map_hostname] = map_ipaddr;
+	    }
+/*duplicated hostname*/
+	  else if (!strcmp (user_host_Map.find (map_hostname)->second.c_str (), hostent_List[host_count].ipaddr))
+	    {
+/*duplicated hostname and ip address both*/
+	    }
+	  else
+	    {
+/*duplicated hostname but different ip address*/
+	      return NULL;
+	    }
+	  host_count++;
+	}
+    }
+  return host_count;
+}
+
+/*
+ * gethostbyname_uhost () - Do same job with gethostbyname (), using by the 'user' defined 'hosts.conf' file.
+ * 
+ * return   : the hostent pointer.
+ * hostname (in) : the hostname.
+ *
+ */
+struct hostent *
+gethostbyname_uhost (char *name)
+{
+  struct hostent *hp;
+
+  if (host_conf_Use < 0)
+    {
+      host_conf_Use = prm_get_bool_value (PRM_ID_USE_USER_HOSTS);
+    }
+
+  if (host_conf_Use == 0)
+    {
+      return gethostbyname (name);
+    }
+
+  hp = host_lookup_internal (name, NULL, HOSTNAME_TO_IPADDR);
+
+  if (hp == NULL)
+    {
+      return NULL;
+    }
+
+  return hp;
+}
+
+//gethostbyname_r_uhost (const char *hostname, struct hostent *out_hp)
+#ifdef HAVE_GETHOSTBYNAME_R
+#if defined (HAVE_GETHOSTBYNAME_R_GLIBC)
+int
+gethostbyname_r_uhost (const char *name,
+		       struct hostent *ret, char *buf, size_t buflen, struct hostent **result, int *h_errnop)
+#elif defined (HAVE_GETHOSTBYNAME_R_SOLARIS)
+struct hostent *
+gethostbyname_r_uhost (const char *name, struct hostent *ret, char *buf, size_t buflen, int *h_errnop)
+#elif defined (HAVE_GETHOSTBYNAME_R_HOSTENT_DATA)
+int
+gethostbyname_r_uhost (const char *name, struct hostent *ret, struct hostent_data *ht_data)
+#else
+struct hostent *
+gethostbyname (const char *name)
+#error "HAVE_GETHOSTBYNAME_R"
+#endif				/* HAVE_GETHOSTBYNAME_R_GLIBC */
+#endif				/* HAVE_GETHOSTBYNAME_R */
```

Comment:
if HAVE_GETHOSTBYNAME_R is not defined, function body (line# 293 ~ 349) should not be defined also.
So, '#endif' for HAVE_GETHOSTBYNAME_R should be move to line #350.
It causes build error in windows.

Reply:
Oh! Thank you! Actually, I tried to solve this problem several times but I couldn't. 

So I try to ask you this problem with bundle questions. 

I will fix it!

Reply:
It does not work. So I will make meaningless function to avoid Windows error.